File: frontend/src/app/service/site.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { Site } from '../model/site.interface';
import { PageResponse } from './usuario.service';

@Injectable({
  providedIn: 'root'
})
export class SiteService {
  private apiUrl = 'http://localhost:8080/api/site'; // Ajusta la URL según tu backend

  constructor(private http: HttpClient) {}

  // Crear o actualizar site
  guardar(site: Site): Observable<Site> {
    if (site.idSite) {
      // Si tiene ID, es una actualización
      return this.http.put<Site>(`${this.apiUrl}/${site.idSite}`, site);
    } else {
      // Si no tiene ID, es una creación
      return this.http.post<Site>(this.apiUrl, site);
    }
  }

  // Listar con paginación y filtros
  listar(
    page: number = 0,
    size: number = 10,
    sort: string = 'codigoSitio',
    direction: string = 'asc',
    activos?: boolean
  ): Observable<PageResponse<Site>> {
    let params = new HttpParams()
      .set('page', page.toString())
      .set('size', size.toString())
      .set('sort', sort)
      .set('direction', direction);

    if (activos !== undefined) {
      params = params.set('activos', activos.toString());
    }

    return this.http.get<PageResponse<Site>>(this.apiUrl, { params });
  }

  // Buscar sites con texto
  buscar(search: string, page: number = 0, size: number = 10): Observable<PageResponse<Site>> {
    let params = new HttpParams()
      .set('search', search)
      .set('page', page.toString())
      .set('size', size.toString());

    return this.http.get<PageResponse<Site>>(`${this.apiUrl}/buscar`, { params });
  }

  // Obtener site por ID
  obtenerPorId(id: number): Observable<Site> {
    return this.http.get<Site>(`${this.apiUrl}/${id}`);
  }

  // Toggle activo/inactivo
  toggle(id: number): Observable<void> {
    return this.http.post<void>(`${this.apiUrl}/${id}/toggle`, {});
  }

  // Eliminar site (soft delete) - usa el mismo toggle
  eliminar(id?: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`);
  }
}

File: comfutura/src/main/java/com/backend/comfutura/service/SiteService.java
package com.backend.comfutura.service;

import com.backend.comfutura.dto.Page.PageResponseDTO;
import com.backend.comfutura.model.Site;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface SiteService {

    // Paginados con PageResponseDTO
    PageResponseDTO<Site> listar(Pageable pageable);
    PageResponseDTO<Site> listarActivos(Pageable pageable);
    PageResponseDTO<Site> buscar(String search, Pageable pageable);

    // Para mantener compatibilidad
    Page<Site> listarPaginado(Pageable pageable);

    // CRUD
    Site guardar(Site site);
    Site obtenerPorId(Integer id);
    void toggle(Integer id);

    // Métodos de validación
    boolean existeCodigoSitio(String codigoSitio);
    boolean existeCodigoSitioConOtroId(String codigoSitio, Integer id);
}